Algebra.java

```diff
@@ -26,42 +26,178 @@ public static void main(String args[]) {
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
 		// Replace the following statement with your code
-		return 0;
+		int sum = x1;
+		if (x2>=0)
+		{
+			for (int i=0; i<x2;i++)
+			{
+				sum++;
+			}
+		}
+		else
+		{
+			for ( int j=0; j>x2;j--)
+		{
+			sum--;
+		}
+		}
+		return sum;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
 		// Replace the following statement with your code
-		return 0;
+		int sum = x1;
+		if (x2>=0)
+		{
+			for ( int i=0; i<x2;i++)
+			{
+				sum--;
+			}
+		}
+		else
+		{
+			for (int j=0; j>x2;j--)
+			{
+				sum++;
+			}
+		}
+		return sum;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
 		// Replace the following statement with your code
-		return 0;
+		int sum = 0;
+		if(x1==0 || x2==0)
+		{
+			return 0 ;
+		}
+		if(x1>0&&x2>0)
```

Comment:
You checked every possible combination of `x1` and `x2`, but there is a much short and general conditions checking approach...

Try to think about that.